server.ts

```diff
@@ -67,6 +74,30 @@ app.get("/s/", async (req, res) => {
     console.error(error);
     res.redirect("/");
   }
+}
+// Test URL: https://raw.githubusercontent.com/Watts-Lab/surveyor/main/surveys/CRT.csv
+// e.g. http://localhost:4000/s/?url=https://raw.githubusercontent.com/Watts-Lab/surveyor/main/surveys/CRT.csv&name=Mark
+app.get("/s/", async (req, res) => {
+
+  //For debugging purposes. Prints encrypted version of url. In the future, this will be done in researcher menu.
+  const cipher = crypto.createCipheriv(crypto_algorithm, private_key_example, iv_example);
+  let encrypted = cipher.update(String(req.query.url), 'utf8', 'hex');
+  console.log(encrypted += cipher.final('hex'));
+
+  getsurvey(String(req.query.url), req, res);
+});
+
+app.get('/e/:data', async (req, res) => {
+  //in the future, private_key and iv will be obtained through researcher database
+  try {
+    const decipher = crypto.createDecipheriv(crypto_algorithm, private_key_example, iv_example);
+    let decrypted = decipher.update(req.params.data, 'hex', 'utf8');
+    decrypted += decipher.final('utf8');
```

Comment:
Ah, we also need something to identify the researcher, perhaps `/e/:data/:researcher_id` so that we decrypt correctly. 

Perhaps we can assume that if `:researcher_id` is blank it is us (and just use the random numbers as you have it) (until the researcher db step is done, which it sounds like will be next.

Reply:
I suppose an even more adaptable version of this would accept either encrypted queries or encrypted URLs + non encrypted queries, and just bundle everything from the encrypted or non encrypted queries into one object that we use from then on.

Reply:
I have changed the `decryption` branch so that server.ts will encrypt the full query instead of just the URL. I will add the researcherID param to /e/ once I get the researcher database example running (in the `sampleresearcherdb` branch)